app/views/events/index.html.erb

```diff
@@ -1,48 +1,33 @@
+<% display_no_ttt_events_message = display_no_ttt_events_message?(@performed_search, @events_by_type, @events_search_type) %>
+
 <% @search_component = Events::SearchComponent.new(
   @event_search,
   search_events_path(anchor: "searchforevents")
 ) %>
 
 <% @before_search_fullwidth = true %>
 
-<% content_for :before_search do %>
-<h2>Types of events</h2>
-<section class="event-type-descriptions">
-  <section class="event-type-descriptions__content">
-    <% GetIntoTeachingApiClient::Constants::EVENT_TYPES.values.each do |type_id| %>
-      <%= render Events::TypeDescriptionComponent.new(type_id) %>
-    <% end %>
-  </section>
-</section>
-<% end %>
-
-<% unless @performed_search %>
-  <div class="content-alert content-alert--left">
-    <div class="content-alert__icon">
-      <%= image_pack_tag "media/images/icon-moved-online-purple.svg", alt: "", size: "40x26" %>
-    </div>
-    <div>
-      <h4>Some events have moved online</h4>
-      <p>Until it is safe to see you in person, the Train to Teach and most of School and University events are online.</p>
-    </div>
-  </div>
+<% if display_no_ttt_events_message %>
+  <%= render(Events::NoResultsComponent.new) do %>
+    <p>DfE's Train to Teach events will return in September. Sign up for updates to be notified about spaces.</p>
+    <%= link_to('Sign up for notifications', mailing_list_steps_url, class: "button") %>
+  <% end %>
 <% end %>
 
 <% unless @no_results %>
-  <% unless @performed_search %>
-  <h2>Discover events</h2>
-  <% end %>
   <% @events_by_type.each do |type_id, events| %>
     <%= render "event_group",
       type_id: type_id,
       events: events,
       page_param_name: @performed_search ? @group_presenter.page_param_name(type_id) : nil,
-      show_see_all_events: true
+      show_see_all_events: events.blank? && type_id == 222750001 ? false : true
```

Comment:
Determined to get that ternary in aren't you? 😅 

```
show_see_all_events: events.present? && type_id != 222750001
```

Worth having the integer as a constant as its not clear which type it is? same for the conditional below

Reply:
> Determined to get that ternary in aren't you? 

Hahaha, I hope I'm not wrong this time 
 
> ```
> show_see_all_events: events.present? && type_id != 222750001
> ```
So, the `events.present? && type_id != 222750001` will evaluate to false and hide the "see all events" button when there are events to show in the TTT block, but what we want is to hide the button only when there are no events to show in the TTT block. 

This table may help to illustrate the condition
TTT category | Has no events | Hide the button
-- | -- | --
TRUE | TRUE | FALSE
TRUE | FALSE | TRUE
FALSE | TRUE | FALSE
FALSE | FALSE | FALSE

May worth exploring the idea of hiding the button in any category (not just TTT) that doesn't have events. In another iteration though as that would be out of this PR's scope.

> Worth having the integer as a constant as its not clear which type it is? same for the conditional below

Actually, we do have the `GetIntoTeachingApiClient::Constants::EVENT_TYPES["Train to Teach event"]`. I could use that instead.

Reply:
The table definitely helps. Perhaps it should be in a helper method with an explanation of the logic/reasoning?

Reply:
Ah forgot to switch it to an or when I negated it, so I think this would work:

```
show_see_all_events: events.present? || type_id != 222750001
```

But yeah as Pete mentioned a helper is probably the way to go

Reply:
>The table definitely helps. Perhaps it should be in a helper method with an explanation of the logic/reasoning?

Good call. I've moved it in the `show_see_all_events_button?()` helper method and added some comments. I hope it's more clear now.

> > Worth having the integer as a constant as its not clear which type it is? same for the conditional below
>
> Actually, we do have the GetIntoTeachingApiClient::Constants::EVENT_TYPES["Train to Teach event"]. I could use that instead.

I replaced that code with the `ttt_event_type_id()` helper method.